docs/blueprints/deletion-performance.md

```diff
@@ -0,0 +1,170 @@
+- Storage nodes will have more garbage if we use a connectionless transport protocol
+- Storage nodes will have more garbage if we use a probabilistic approach.
+
+
+### (2) Satellite delete pieces of the storage nodes reliably
+
+Uplink will communicate with the satellite as it currently does.
+
+The satellite will take care of communicating with the storage nodes for deleting the pieces using RPC.
+
+Advantages:
+
+- Uplink deletion operation will be independent of the size of the file, guaranteeing always being responsive.
+- It doesn't present a risk of leaving garbage in the storage nodes when deletion operation is interrupted.
+- In general, the storage nodes will have less garbage because of deletions.
+
+Disadvantages:
+
+- The satellite requires a new chore to delete the pieces of the storage nodes. The increment of network traffic, computation, and data to track the segments to delete will increase the running costs.
+- The satellite will have another component incrementing the cost of the operation as monitoring, replication, etc.
+
+
+### (3) Satellite delete pieces of the storage nodes unreliably
+
+Uplink will communicate with the satellite as it currently does.
+
+The satellite will take care of communicating with the storage nodes for deleting the pieces using a connectionless protocol like UDP.
+
+Advantages:
+
+- Uplink deletion operation will be independent of the size of the file, guaranteeing always being responsive.
+- It doesn't present a risk of leaving garbage in the storage nodes when deletion operation is interrupted.
+
+Disadvantages:
+
+- The satellite requires a new chore to delete the pieces of the storage nodes. The increment of network traffic, computation, and data to track the segments to delete will increase the running costs.
+- The satellite will have another component incrementing the cost of the operation as monitoring, replication, etc.
+
+### Conclusion
+
+The alternative approach (1):
+
+- It is similar to the current mechanism but with some improvements towards the goals.
+- It doesn't add more load to the satellite, but we cannot trust in the uplink in deleting the pieces or informing the non-deleted ones.
+
+The alternative approaches (2) and (3) are similar.
+
+Approach (2) has the advantage of guaranteeing less garbage left on the storage nodes at the expense of requiring more network traffic.
+
+Approach (3) requires less network traffic, but it may not require less computation considering that we may need to encrypt the data sent through UDP.
+
+Approach (3) may get rid of the satellite garbage faster.
+
+Both approaches, (2) and (3), present the problem of increasing garbage on the satellite side.
+
+Taking one of these approaches will require a study on how to keep the less amount of garbage as possible.
+
+## Implementation
+
+1. Adapt protocol buffers definitions for storage node delete operations.
```

Comment:
Can we describe here how the protobufs should be changed? So we can discuss any issues with backward-compatibility?

Reply:
I'll take a look because I'm not sure if it has to change or not because I don't know anything about this part.

Reply:
I've restructured a bit the implementation section to mention the changes in the protocol buffers and I added an issue for discussing them because of the possible backward-incompatibility.